### ISSUE TITLE
training accuracy

### ISSUE THREAD

**Issue by Araik-77** [jeffsonyu/VTacO#5]
thank you for your replay, I can run this program now. However, I can not get the same result in your paper. It may be my fault.
For example, I want to trian IoU of **box**, I first run

`python train_depth.py configs/tactile/tactile_test.yaml`

I get a loss about "0.15" after 300 epochs, and then run

`python train.py configs/VTacO/VTacO_YCB.yaml`

I can only get IoU of **box** about "0.8", which is not equal as "0.950" in your paper,

![2023-06-20 15-52-56屏幕截图](https://github.com/jeffsonyu/VTacO/assets/52837802/030b74b8-3311-4f63-b8ac-a83f5174a540)

Below is a part of YCB configs, I did not modify anything. If I need to change model "train_tactile" or "with_contact"  to "true"
![2023-06-20 15-54-33屏幕截图](https://github.com/jeffsonyu/VTacO/assets/52837802/74e9af01-2e8a-47b0-a88e-faa867108635)

**I may miss some details, so could you give a guidance for how to train the model.**

**Comment by jeffsonyu** [comment]
Sorry for the late reply, we were on vacation. We updated our code 20 days ago, maybe you can try the updated repo to train it again? I apologize for our negligence, and our code will be continuously updated and improved.

**Comment by Araik-77** [comment]
Forgive me for not interrupting your vacation. 
I meet some trouble to repeat your results. I have used the latest repo to train.
**1**. I loaded the datasets from  [https://huggingface.co/datasets/robotflow/vtaco](datasets), however, the"val.lst" has some data that not included in file. such as, "**foldingrack09_053_0002,foldingrack09_053_0004,foldingrack09_053_0006,foldingrack09_053_0008**".
 The first picture is a screenshot of the data directory and the second picture is a screenshot of val.list. 
**the same thing happened** in "scissor02_009_, lock01_002_, ", If the datasets missing when uploading？
![2023-07-03 15-23-20屏幕截图](https://github.com/jeffsonyu/VTacO/assets/52837802/e5f929b3-4c8a-4694-8e62-3e17f82bd39c)


![2023-07-03 15-27-44屏幕截图](https://github.com/jeffsonyu/VTacO/assets/52837802/3eca35d5-a012-4bf1-b87b-a4a0d8d42796)


**2**.I removed the "foldingrack09_053_0002,foldingrack09_053_0004,foldingrack09_053_0006,foldingrack09_053_0008" from the val.lst, and now I can run the program. However, in VTacO,I get 0.5732 IoU of foldingrack, while 0.782 in your paper. I follow the hyperparameters in your paper , but I do not understand 
> For the training of VTacO, we set the batch size as 6 and LR = 2e − 4 with Adam optimizer to train on the whole simulated dataset for 100 epochs, and finetune on every type of object with batch size 4 and LR = 1e − 4 for50 to 100 epochs according to different types.

if I need to train the whole datasets and using pretrained model to initialize , and then train different types? Or train different types directly.

**3**. I attempt to train foldingrack cd in VTacO, I modify   
  model_selection_metric: cd    ## **iou->cd**
  model_selection_mode: minimize   ##**maximize->minimize**
  and other configs are same as training iou, but I get a keyerror
![2023-07-03 16-00-04屏幕截图](https://github.com/jeffsonyu/VTacO/assets/52837802/0519cd5d-c2bb-41f0-a9b3-b47d8220c3d2)
![2023-07-03 16-04-02屏幕截图](https://github.com/jeffsonyu/VTacO/assets/52837802/6cacc427-102a-47a4-b58c-f04fe7919834)


**It is diffucult to follow your job without your guidance，so，could you update on how to train "cd", "emd" . Gaving the training details to improve accuracy .Or update configs of different types.**

**LOOKING FORWARD TO YOUR REPLY.**

**Comment by jeffsonyu** [comment]
Sorry for the late reply.
1. It's the error of the lst files, we will update it recently. Thank you for noticing!
2. We use the exact same configuration as in the repo, and changed nothing. For instance, we provide a log file of tensorboard for training VTacO on the YCB objects. The file link: [https://drive.google.com/file/d/1mv8ClWa5CcH5_3hC733sI1boXPjhV5xt/view?usp=drive_link](url)
The general way of our training, is to first train the model on the whole dataset, and finetune it on different classes. You can train on the whole dataset by putting every item in each class together and create your own **train.lst**, **test.lst** and **val.lst** for training, and then finetune on each class. You can still train every class from scratch for testing.
3. We haven't support the training using chamfer distance and earth moving distance. These two are for evaluation. We will possibly add them to the training.
Thank you for supporting our work and let me know if you have other questions.

**Comment by jeffsonyu** [comment]
close since no update.
